core/trace/dt_span_exporter.go

```diff
@@ -153,6 +142,18 @@ func (e *dtSpanExporterImpl) newRequest(ctx context.Context, body io.Reader) (*h
 	req.Header.Set("User-Agent", fmt.Sprintf("odin-go/%s %#016x %s",
 		version.FullVersion, e.config.AgentId, e.config.Tenant))
 	req.Header.Set("Accept", "*/*; q=0")
+	// Setting just the header Idempotency-Key with an empty value ensures that the request is
+	// treated as idempotent but the header is not sent over the wire. See net/http/transport.go
+	req.Header.Set("Idempotency-Key", "")
+
+	// GetBody must be set to allow for automatic retries. See net/http/transport.go
+	req.GetBody = func() (io.ReadCloser, error) {
```

Comment:
What is the error parameter here? Should we debug-log it?

Reply:
It's the (second) return value, not a parameter :)

We could log the potential error returned from `body.Seek`, but that should always be nil anyway (both parameters which can cause errors are compile time constants and should never cause said error).

Reply:
Ah, sorry, looks like I'm still not very used to reading Go!